atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepableTimestampsTest.java

```diff
@@ -78,73 +79,83 @@ public void cannotReadForWrongShard() {
+        assertThat(sweepTs).isGreaterThan(TS);
+        assertThat(readConservative(shardCons)).contains(TS_FINE_PARTITION);
     }
 
     @Test
-    public void canReadNextTimestampWhenSweepTimestampInNextPartitionAndSameBucket() {
-        immutableTs = TS + TS_FINE_GRANULARITY - TS % TS_FINE_GRANULARITY;
+    public void noNextTimestampWhenWhenSweepTimestampInSamePartitionAndLower() {
+        long sweepTs = setSweepTimestampAndGet(minForFinePartition(TS_FINE_PARTITION));
 
-        assertThat(tsPartitionFine(immutableTs)).isGreaterThan(tsPartitionFine(TS));
-        assertThat(tsPartitionCoarse(immutableTs)).isEqualTo(tsPartitionCoarse(TS));
-        assertThat(readConservative(shardCons)).contains(TS_REF);
+        assertThat(tsPartitionFine(sweepTs)).isEqualTo(TS_FINE_PARTITION);
+        assertThat(sweepTs).isLessThan(TS);
+        assertThat(readConservative(shardCons)).isEmpty();
     }
 
     @Test
     public void canReadNextIfNotProgressedBeyondForConservative() {
-        progress.updateLastSweptTimestampPartition(thorough(shardCons), TS_REF);
-        assertThat(readConservative(shardCons)).contains(TS_REF);
+        progress.updateLastSweptTimestamp(thorough(shardCons), TS);
+        progress.updateLastSweptTimestamp(conservative(shardThor), TS);
+        progress.updateLastSweptTimestamp(thorough(shardThor), TS);
+        progress.updateLastSweptTimestamp(conservative(shardCons), TS - 1);
+
+        assertThat(readConservative(shardCons)).contains(TS_FINE_PARTITION);
+    }
 
-        progress.updateLastSweptTimestampPartition(conservative(shardCons), TS_REF - 1);
-        assertThat(readConservative(shardCons)).contains(TS_REF);
-        assertThat(readThorough(shardThor)).contains(TS2_REF);
+    @Test
+    public void canReadNextTimestampIfProgressedBeyondButInSamePartitionForConservative() {
+        progress.updateLastSweptTimestamp(conservative(shardCons), maxForFinePartition(TS_FINE_PARTITION) - 1);
+        assertThat(readConservative(shardCons)).contains(TS_FINE_PARTITION);
     }
 
     @Test
-    public void noNextTimestampIfProgressedBeyondForConservative() {
-        progress.updateLastSweptTimestampPartition(conservative(shardCons), TS_REF);
+    public void noNextTimestampIfProgressedToEndOfPartitionForConservative() {
+        progress.updateLastSweptTimestamp(conservative(shardCons), maxForFinePartition(TS_FINE_PARTITION));
         assertThat(readConservative(shardCons)).isEmpty();
-        assertThat(readThorough(shardThor)).contains(TS2_REF);
     }
 
     @Test
     public void canReadNextIfNotProgressedBeyondForThorough() {
-        progress.updateLastSweptTimestampPartition(conservative(shardThor), TS2_REF);
-        assertThat(readThorough(shardThor)).contains(TS2_REF);
+        progress.updateLastSweptTimestamp(thorough(shardCons), TS2);
+        progress.updateLastSweptTimestamp(conservative(shardThor), TS2);
+        progress.updateLastSweptTimestamp(thorough(shardThor), TS2 - 1);
```

Comment:
The mixtures of thorough and conservative are confusing here - I can't tell what each step is supposed to do.

Reply:
This made sense in my head, but doesn't anymore. I'll refactor it into two tests that actually do make sense

Reply:
Also the test was bad, now should make more sense